src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -525,7 +525,9 @@ protected override Match<SyntaxNode> ComputeBodyMatch(SyntaxNode oldBody, Syntax
                 // - for lambdas the root is a LambdaExpression.
                 // - for query lambdas the root is the query clause containing the lambda (e.g. where).
 
-                return new StatementSyntaxComparer(oldBody, newBody).ComputeMatch(oldBody.Parent, newBody.Parent, knownMatches);
+                var oldLambda = GetLambda(oldBody);
```

Comment:
>GetLambda [](start = 32, length = 9)

This can't work. GetLambda asserts that the result is a lambda, which is not always the case

Reply:
We want the root to be ArrowExpressionClauseSyntax here. Why do we need to change this?

---
In reply to: [143808933](https://github.com/dotnet/roslyn/pull/22624#discussion_r143808933) [](ancestors = 143808933)

Reply:
Sorry for the first version! I think I sent it too early.

I do not like the way ComputeBodyMatch makes matches. Actually, it matches "int local() { return 1; }" with "=> 1". However, as I found later, this happens not only for local functions but for regular methods too. I think there is a inconsistency when matching expression bodies and block bodied members. We can discuss offline if we want to make matches consistent.

So, it seems that there is an actual code problem on the compiler level only. And I have attempted to fix it with a new iteration.

---
In reply to: [143809607](https://github.com/dotnet/roslyn/pull/22624#discussion_r143809607) [](ancestors = 143809607,143808933)

Reply:
The matching root choice was very deliberate, at least for lambdas and method bodies.